common/network/IPV6Address.cpp

```diff
@@ -0,0 +1,172 @@
+/*
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ * IPV6Address.cpp
+ * An IPV6 address
+ * Copyright (C) 2023 Peter Newman
+ */
+
+#if HAVE_CONFIG_H
+#include <config.h>
+#endif  // HAVE_CONFIG_H
+
+#ifdef HAVE_SYS_SOCKET_H
+#include <sys/socket.h>  // Required by FreeBSD
+#endif  // HAVE_SYS_SOCKET_H
+#ifdef HAVE_ARPA_INET_H
+#include <arpa/inet.h>
+#endif  // HAVE_ARPA_INET_H
+#ifdef HAVE_NETINET_IN_H
+#include <netinet/in.h>  // Required by FreeBSD
+#endif  // HAVE_NETINET_IN_H
+
+#include <assert.h>
+#include <math.h>
+#include <stdint.h>
+#include <limits>
+#include <string>
+
+#include "common/network/NetworkUtilsInternal.h"
+#include "ola/Logging.h"
+#include "ola/network/IPV6Address.h"
+#include "ola/network/NetworkUtils.h"
+
+namespace ola {
+namespace network {
+
+using std::string;
+
+IPV6Address::IPV6Address(const uint8_t *address) {
+  // TODO(Peter): Deal with any network byte order conversion?
+  memcpy(&m_address.s6_addr[0], address, sizeof (struct in6_addr));
+}
+
+bool IPV6Address::operator<(const IPV6Address &other) const {
+  // TODO(Peter): Deal with any network byte order conversion?
```

Comment:
can this TODO comment be removed? The next line seems to clarify it?

Reply:
TBH the next line is a copy from the IPv4 function. Maybe I should make it more clear the answer is unknown.

Currently these classes are entirely standalone, I've only shuffled bytes into them in the order I've chosen and got the results I expected, the real test (which needs more code) will be to actually read something from a machine (or over the network) and confirm we still get the right result. And then repeat on a BE machine...

Reply:
Actually it's worse than that, I'll just drop the invalid comment on the line below for now:
https://github.com/OpenLightingProject/ola/blob/87b1318ded6947d78be62ee8fda429305558f94b/common/network/IPV4Address.cpp#L58-L66

Reply:
Okay false comment now removed, I'll get this one merged as there are related bits backing up behind it, can/will review/reconsider when it does something more than just shuffling bytes in and out of other functions...